python/trezorlib/tests/device_tests/test_msg_eos_signtx.py

```diff
@@ -0,0 +1,1132 @@
+# This file is part of the Trezor project.
+#
+# Copyright (C) 2012-2018 SatoshiLabs and contributors
+#
+# This library is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License version 3
+# as published by the Free Software Foundation.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the License along with this library.
+# If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.
+
+import pytest
+import time
+import json
+
+from collections import namedtuple
+from trezorlib import eos, messages as proto
+from trezorlib.tools import parse_path
+from trezorlib.eos import parse_transaction_json, parse_action
+from .common import TrezorTest
+
+
+@pytest.mark.eos
+class TestMsgEosSignTx(TrezorTest):
+    def test_eos_signtx_transfer_token(self):
+        self.setup_mnemonic_nopin_nopassphrase()
+        data = '''{
+            "chain_id": "cf057bbfb72640471fd910bcb67639c22df9f92470936cddc1ade0e2f2e7dc4f",
+            "transaction": {
+            "expiration": "2018-07-14T10:43:28",
+            "ref_block_num": 6439,
+            "ref_block_prefix": 2995713264,
+            "net_usage_words": 0,
+            "max_cpu_usage_ms": 0,
+            "delay_sec": 0,
+            "context_free_actions": [],
+            "actions": [
+                {
+                "account": "eosio.token",
+                "name": "transfer",
+                "authorization": [
+                    {
+                    "actor": "miniminimini",
+                    "permission": "active"
+                    }
+                ],
+                "data": {
+                    "from": "miniminimini",
+                    "to": "maximaximaxi",
+                    "quantity": "1.0000 EOS",
+                    "memo": "testtest"
+                }
+                }
+            ],
+            "transaction_extensions": []
+            }
+        }'''
+
+        tx = parse_transaction_json(json.loads(data))
+        self.client.transport.write(_tx_msg(tx))
+        response = self.client.transport.read()
+
+        assert isinstance(response, proto.ButtonRequest)
```

Comment:
instead of doing this, make use of input flow functionality. see e.g. NEM or Cardano tests for examples

in here i'd do something like this:
```python
def input_flow(self, num_swipes):
    # confirm number of actions
    yield
    self.client.debug.press_yes()
    # swipe through pages
    time.sleep(1)
    for _ in range(num_swipes):
        self.client.debug.swipe_down()
        time.sleep(1)
    # confirm last page
    self.client.debug.press_yes()
```

then on usage:
```python
with self.client:
    self.client.set_input_flow(self.input_flow(num_swipes=3))
    resp = eos.sign_tx(self.client, ADDRESS_N, CHAIN_ID, transaction)
    assert resp.signature_x == ...
```

Reply:
Thanks for good explanation and code snippets below. I think now it should be alright. It takes some time to get familiar with the changes and to fully understand how stuff works and of course therefore make some adequate changes :)